src/components/SidebarList/SidebarList.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import ShowMore from './ShowMore';
+import Item from './SidebarListItem';
+
+const ListContainer = styled.ul`
+  display: flex;
+  flex-direction: column;
+  list-style-type: none;
+  margin: 0;
+  padding: 0;
+
+  &::-webkit-scrollbar-thumb {
+    background: ${({ theme }) => theme.colors.gutter};
+  }
+
+  &::-webkit-scrollbar-track {
+    background: ${({ theme }) => theme.colors.grayDark};
+  }
+`;
+
+class SidebarList extends React.Component {
+  static propTypes = {
+    selectedIndex: PropTypes.number,
+    children: PropTypes.node.isRequired,
+    hasNextPage: PropTypes.bool,
+    hasPreviousPage: PropTypes.bool,
+    onNextPageClicked: PropTypes.func,
+    onPreviousPageClicked: PropTypes.func,
+  };
+
+  static defaultProps = {
+    selectedIndex: -1,
+    hasNextPage: false,
+    hasPreviousPage: false,
+    onNextPageClicked: () => null,
+    onPreviousPageClicked: () => null,
+  };
+
+  renderPrevious = () => {
+    const { hasPreviousPage, onPreviousPageClicked } = this.props;
+    if (hasPreviousPage) {
+      return <ShowMore onClick={onPreviousPageClicked}>Show Previous</ShowMore>;
+    }
+
+    return null;
+  };
+
+  renderNext = () => {
+    const { hasNextPage, onNextPageClicked } = this.props;
+    if (hasNextPage) {
+      return <ShowMore onClick={onNextPageClicked}>Show Next</ShowMore>;
+    }
+
+    return null;
+  };
+
+  renderItems = () => {
+    const { selectedIndex } = this.props;
+
+    return React.Children.map(
+      this.props.children.filter((child) => child !== null),
```

Comment:
`this.props.children.filter` relies on type array however if there is only a single child element then `this.props.children` is set to the react element itself / a factory function. This will throw an error on an undefined method.